internal/ansible/watches/watches.go

```diff
@@ -358,6 +362,32 @@ func Load(path string, maxReconciler, ansibleVerbosity int) ([]Watch, error) {
 	return watches, nil
 }
 
+// replaceEnvVariables will replace all ${VAR} references found in the byte array
+// with the actual values of the env variables. If an env variable is not defined
+// the ${VAR} reference remains as-is in the returned byte array.
+func replaceEnvVariables(input []byte) []byte {
+	// Regular expression to match "${VARIABLE}"
+	envVarPattern := regexp.MustCompile(`\${([^}]+)}`)
```

Comment:
I'm a bit concerned about using/maintaining regex here. I wonder if there is a good way for us to use go templates in the yaml as opposed to finding regex matches for "${...}".

I haven't had much time to dig into this thought more, but I can play around with some stuff tomorrow and see if we can get something a bit more robust

Reply:
What's the concern over using a regex pattern here? It's a fairly innocuous regex pattern that is just picking out the `${var}` syntax. It's a pretty typical regex pattern that would likely never have to change (certainly that regex pattern is not anything out of the ordinary).

Reply:
@everettraven check out the latest commit - removes regex dependency. Simply uses os.Expand functionality instead.

Reply:
> What's the concern over using a regex pattern here? It's fairly innocuous that is just picking out the `${var}` syntax. It's a pretty typical pattern that would likely never have to change (certainly that regex pattern is not anything out of the ordinary).

My concern is simply maintaining the regex. In my experience using custom regex over other methods can result in unexpected behavior and bugs. For example, this regex currently matches all the following:
- ${SOMETHING}
- ${SOMETHING_COOL}
- ${NICE-REG}
- ${{{{Some cool thing}}
- ${5Things}
- ${TheThing=AnotherThing}

I'm by no means an expert in environment variable naming and what is allowed and what is not but some of these seem like they would be invalid (please do correct me if I'm wrong!). For every one of these edge cases the regex syntax gets more and more complex.

Using the os.Expand functionality is an approach I'm personally more comfortable with, especially because it is part of the standard library.